src/zarr/core/metadata/v2.py

```diff
@@ -140,6 +141,22 @@ def from_dict(cls, data: dict[str, Any]) -> ArrayV2Metadata:
         _data = data.copy()
         # check that the zarr_format attribute is correct
         _ = parse_zarr_format(_data.pop("zarr_format"))
+        dtype = parse_dtype(_data["dtype"])
+
+        if dtype.kind in "SV":
+            fill_value_encoded = _data.get("fill_value")
+            if fill_value_encoded is not None:
+                if dtype.kind == "S":
+                    try:
+                        fill_value = base64.standard_b64decode(fill_value_encoded)
+                        _data["fill_value"] = fill_value
+                    except Exception:
```

Comment:
is it possible to limit the exceptions we catch here to only those that could be raised by `standard_b64decode`?

Reply:
Yes, this came from the old implementation.

While writing a test, I found that we just raise an exception later on in `parse_fill_value` so AFAICT there's no value in trying to suppress exceptions here.